gazelle/docs/installation_and_usage.md

```diff
@@ -0,0 +1,156 @@
+# Installation and Usage
+
+## Example
+
+Examples of using Gazelle with Python can be found in the `rules_python`
+repo:
+
+* bzlmod: {gh-path}`examples/bzlmod_build_file_generation`
+* WORKSPACE: {gh-path}`examples/build_file_generation`
+
+:::{note}
+The following documentation covers using bzlmod.
+:::
+
+
+## Adding Gazelle to your project
+
+First, you'll need to add Gazelle to your `MODULE.bazel` file. Get the current
+version of Gazelle from their [releases page][gazelle-releases].
+
+[gazelle-releases]: https://github.com/bazel-contrib/bazel-gazelle/releases/
+
+See the installation `MODULE.bazel` snippet on the `rules_python`
+[releases page][rules-python-releases] in order to configure `rules_python`.
+
+[rules-python-releases]: https://github.com/bazel-contrib/rules_python/releases
+
+You will also need to add the {bzl:obj}`bazel_dep` for configuration for
+`rules_python_gazelle_plugin`.
+
+Here is a snippet of a `MODULE.bazel` file.
+
+```starlark
+# The following stanza defines the dependency rules_python.
+bazel_dep(name = "rules_python", version = "0.22.0")
+
+# The following stanza defines the dependency rules_python_gazelle_plugin.
+# For typical setups you set the version.
+bazel_dep(name = "rules_python_gazelle_plugin", version = "0.22.0")
+
+# The following stanza defines the dependency gazelle.
+bazel_dep(name = "gazelle", version = "0.31.0", repo_name = "bazel_gazelle")
+
+# Import the python repositories generated by the given module extension into
+# the scope of the current module.
+use_repo(python, "python3_9")
+use_repo(python, "python3_9_toolchains")
+
+# Register an already-defined toolchain so that Bazel can use it during
+# toolchain resolution.
+register_toolchains(
+    "@python3_9_toolchains//:all",
+)
+
+# Use the pip extension
+pip = use_extension("@rules_python//python:extensions.bzl", "pip")
+
+# Use the extension to call the `pip_repository` rule that invokes `pip`, with
+# `incremental` set.
+# Accepts a locked/compiled requirements file and installs the dependencies listed within.
+# Those dependencies become available in a generated `requirements.bzl` file.
+# You can instead check this `requirements.bzl` file into your repo.
+# Because this project has different requirements for windows vs other
+# operating systems, we have requirements for each.
+pip.parse(
+    name = "pip",
+    requirements_lock = "//:requirements_lock.txt",
+    requirements_windows = "//:requirements_windows.txt",
+)
+
+# Imports the pip toolchain generated by the given module extension into the
+# scope of the current module.
+use_repo(pip, "pip")
+```
+
+Next, we'll fetch metadata about your Python dependencies, so that gazelle can
+determine which package a given import statement comes from. This is provided
+by the {bzl:obj}`modules_mapping` rule. We'll make a target for consuming this
+{bzl:obj}`modules_mapping`, and writing it as a manifest file for Gazelle to read.
```

Comment:
It seems that the links here don't work - we cannot get to the starlark docs to here in the https://rules-python--3132.org.readthedocs.build/en/3132/gazelle/docs/installation_and_usage.html

Do you have a plan how to make it work in the future?

Reply:
The way to do it would be to define the `bzl_library` targets in the `gazelle` workspace and then include them into the `//docs` target. That should fix the crossreffs.

Reply:
POC in https://github.com/bazel-contrib/rules_python/pull/3140; will continue discussion there.